tests/common/widgets/search-box/options.ts

```diff
@@ -0,0 +1,436 @@
+        },
+        widgetParams: { searchAsYouType: false },
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.type(screen.getByRole('searchbox'), 'iPhone');
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.click(
+        screen.getByRole('button', {
+          name: /submit the search query/i,
+        })
+      );
+
+      expect(state.indexName.query).toBe('iPhone');
+    });
+
+    test('resets query when clicking on reset button', async () => {
+      let state: UiState = {};
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+          onStateChange: ({ uiState }) => (state = uiState),
+          initialUiState: {
+            indexName: {
+              query: 'something',
+            },
+          },
+        },
+        widgetParams: {},
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.click(
+        screen.getByRole('button', {
+          name: /clear the search query/i,
+        })
+      );
+
+      expect(state.indexName.query).not.toBeDefined();
+    });
+
+    test('does not update input value when query is updated externally and on focus', async () => {
+      let localSetUiState: InstantSearch['setUiState'];
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+          onStateChange: ({ setUiState }) => (localSetUiState = setUiState),
+        },
+        widgetParams: {},
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.type(screen.getByRole('searchbox'), 'i');
+
+      act(() => {
+        localSetUiState({
+          indexName: {
+            query: 'iPhone',
+          },
+        });
+      });
+
+      expect(screen.getByRole('searchbox')).toHaveValue('i');
+
+      await act(async () => {
+        userEvent.tab();
+
+        await wait(0);
+      });
+
+      expect(screen.getByRole('searchbox')).toHaveValue('iPhone');
+    });
+  });
+
+  skippableDescribe('searchAsYouType option', skippedTests, () => {
+    test('does not refine query on type when `searchAsYouType` is false', async () => {
+      let state: UiState = {};
+      const searchClient = createSearchClient({});
+
+      await setup({
+        instantSearchOptions: {
+          indexName: 'indexName',
+          searchClient,
+          onStateChange: ({ uiState }) => (state = uiState),
+        },
+        widgetParams: { searchAsYouType: false },
+      });
+
+      await act(async () => {
+        await wait(0);
+      });
+
+      userEvent.type(screen.getByRole('searchbox'), 'iPhone');
+
+      expect(state.indexName?.query).not.toBeDefined();
+      expect(screen.getByRole('searchbox')).toHaveValue('iPhone');
+    });
+  });
```

Comment:
What's the difference between this test and the specific one in Vue?

Reply:
There is no `searchAsYouType` option in Vue